lib/Textile/API/index.ts

```diff
@@ -277,15 +278,15 @@ class API {
           const buffer = Buffer.from(result, 'base64')
           const queryEvent = QueryEvent.decode(buffer)
           switch (queryEvent.type) {
-            case 1:
-              cleanup()
-              resolve()
-              break
-            case 0:
+            case QueryEvent.Type.DATA:
               if (queryEvent.data) {
                 handler(queryEvent.data as Contact)
```

Comment:
I know this line isn't part of your PR, but this is wrong. Need to dig down into `queryEvent.data.value` and use `Contact.decode` or whatever.

Reply:
Sure, will fix

Reply:
fixed, added `local` to the handler because this might be useful (local mean the contact already exists)